rust/sbp/src/sbp_string.rs

```diff
@@ -218,6 +307,15 @@ macro_rules! forward_payload_vec {
     };
 }
 
+#[derive(Debug, PartialEq)]
+pub struct UnterminatedError;
```

Comment:
These should all impl Error (and Display). Otherwise they won't compose with things like Box<dyn Error> or anyhow::Error

Reply:
Also should derive `Clone` or else things like `Result<SbpString, UnterminatedError>` won't be cloneable, which can sometimes be annoying